blog-posts/cdk/portfolio/article.md

```diff
@@ -0,0 +1,228 @@
+---
+published: false
+title: 'Easily deploy your portfolio website with AWS CDK 😎'
+cover_image:
+description: ''
+tags: serverless, tutorial, frontend, aws
+canonical_url:
+---
+
+## TL;DR
+
+In this article, I am going to explain how to deploy a portfolio website (can be plain html+css, built react etc...) to the AWS cloud. The aim is to be able to access this website using my own domain name: https://www.pchol.fr, and a HTTPS endpoint, in order to meet the current industry standards.
+
+To reach this goal, I will use the AWS CDK (Cloud Development Kit) combined with Typescript, to provision a 100% "Infrastructure as Code" CloudFront application serving my website all around the world. Here is a quick look of what the architecture of the app will look like at the end:
+
+![CloudFront architecture](./assets/architecture.png 'CloudFront architecture')
+
+From the networking and DNS records to the frontend files hosted online, everything will be defined inside a single Typescript file: this is the power of AWS CDK! I will go through every step of the process with detailed code in this article, but you can find the end product on my [github repository] if needed.
+
+## Set-up a typescript CDK project
+
+_> To use the CDK, you first have to configure an AWS profile in your CLI. The [official documentation][aws credentials doc] is very easy to follow if you need help._
+
+Run these 2 commands in your CLI to setup your CDK project.
+
+```sh
+mkdir portfolio && cd portfolio
+npm i -g cdk
+cdk init --language typescript
+```
+
+`cdk init --language typescript` creates a base typescript repository with the following structure.
+
+```sh
+portfolio
+│   .gitignore
+│   cdk.json
+│   package.json
+│   package-lock.json
+│   ts-config.json
+└───bin
+│   └───portfolio.ts
+└───front
+│   └───src
+│       └───index.html
+│       └───styles.css
+└───lib
+│   └───portfolio-stack.ts
+└───node-modules
+```
+
+- `bin` folder contains the different stacks of your project. This is where I can set environment variables like the AWS accountId or region.
+- `lib` folder contains the details of my stacks. I only have one, and it's where I will provision all the necessary resources to deploy my static website.
+- `cdk.json` contains the global CDK configuration. For this simple use-case, there will be no need to modify it.
+- I manually added the directory `front/src`, it contains my built website (here it's a simple html+css site, but it can be built react or angular for example)
```

Comment:
```suggestion
- I manually added the directory `front/src` which contains my built website (here it's a simple HTML+CSS site, but it can be built with React or Angular for example)
```

ou 

```suggestion
- I manually added the directory `front/src`. It contains my built website (here it's a simple HTML+CSS site, but it can be built with React or Angular for example)
```